ui-charts/src/spaceTimeChart/components/TimeCaptions.tsx

```diff
@@ -72,71 +87,91 @@ const TimeCaptions = () => {
         return false;
       });
 
-      const labelMarkFormatter = (labelLevel: number, i: number) => ({
-        level: labelLevel,
-        rangeIndex: i,
-      });
       const labelMarks = computeVisibleTimeMarkers(
         minT,
         maxT,
         timeRanges,
         labelLevels,
-        labelMarkFormatter
+        (level: number, i: number) => ({
+          level,
+          styles: timeCaptionsStyles[level],
+          formatter: RANGES_FORMATER[i],
+        })
       );
-
+      let allMarksArray = map(labelMarks, (mark, t) => ({
+        ...mark,
+        time: +t,
+      }));
```

Comment:
Since you do this transformation twice (here and line 115), why not move it in `computeVisibleTimeMarkers` ?